app/controllers/dashboards_controller.rb

```diff
@@ -42,14 +42,19 @@ def push
     if version.blank?
       flash.alert = 'No preview version available'
     else
-      pv = Version.create(number: version.number, production: true, user: current_user)
+      production_version = Version.create(number: version.number, production: true, user: current_user)
 
-      if pv.persisted?
+      if production_version.persisted?
         flash.notice = 'Production data updated'
 
         # Build Sitemap and notify search engines in production only
         ping = request.original_url.include?(GibctSiteMapper::PRODUCTION_HOST)
         GibctSiteMapper.new(ping: ping)
+
+        if ENV['ARCHIVE_INSTITUTIONS'] == 'True'
+          # Archive old institution rows
+          InstitutionsArchive.archive(production_version)
```

Comment:
This doesn't have to be done immediately upon creating of a new production `version`.
What if we put it on a [scheduler](https://github.com/jmettraux/rufus-scheduler) to run in the middle of the night?
We could have a script that runs something like this at 4 AM (this assumes my other suggestion that `archive` only archives one version):
```
Version.production.where('number < ?', Version.current_production.number).each do |version|
  InstitutionsArchive.archive(version.number)
end
```
This could be added in a separate PR, after we get the archiving functionality in this PR merged and figure out the best approach to archiving the existing ~250 versions.